src/language-usability.rst

```diff
@@ -16,6 +16,171 @@ Usability of Type Systems
+  objects' constructors on usability. The study in this paper showed that developers
+  strongly preferred and were more effective with APIs that did not
+  require parameters in object constructors. This contradicted the
+  belief that object constructor parameters would "self-document"
+  themselves and guide developers to using the objects correctly.
+
+.. container:: bib-item
+
+  .. bibliography:: api-usability.bib
+    :filter: key == 'stylos2008implications'
+
+  This paper examines the impact of method placement on usability.
+  In the user study participants were given nearly identical APIs but
+  the comparison group contained a method that was placed in an
+  unexpected location. Participants with "correctly-placed" methods
+  were orders of magnitudes faster in their tasks.
+
+Documentation Improving Tools
+-----------------------------
+
+..
+  Outline
+  * API usability can also be improved by creating better tools for understanding and searching API's -- documentation
+  * Another distinction with language design - documentation is part of API. Using API is using documentation
+  * Paper: Jadeite
+  * Paper: Improving API Documentation Usability with Knowledge Pushing
+  * Paper: Mica: A Web-Search Tool for Finding API Components and Examples
+..
+
+API usability can also be improved by creating better tools for understanding and searching through APIs -- improving API documentation
+:cite:`stylos2009improving` :cite:`dekel2009improving` :cite:`stylos2006mica`.
+Documentation represents another distinction of API design from language design.
+The usage of APIs is so closely tied with using the API documentation that
+the documentation can be considered part of the API.
+A good documentation is integral in making an inherently complex API more learnable.
+
+.. container:: bib-item
+
+  .. bibliography:: api-usability.bib
+    :filter: key == 'stylos2009improving'
+
+  This paper presents Jadeite, a tool for improving API documentation by
+  letting API users add "pretend" classes or methods in places where
+  they expect a class or method to exist. This method helps alleviate
+  the problem where classes or methods are located in unexpected locations
+  :cite:`stylos2008implications`.
+
+.. container:: bib-item
+
+  .. bibliography:: api-usability.bib
+    :filter: key == 'dekel2009improving'
+
+  API functions often contain messsages written about rules or caveats to usage.
+  However, these messages can be lost in verbose text, and missed by the API
+  user. This paper shows this phenomenon occurs in real programming, and
+  shows how a function decorator can help alert users of such messages.
+
+.. container:: bib-item
+
+  .. bibliography:: api-usability.bib
+    :filter: key == 'stylos2006mica'
+
+  This paper presents Mica, a tool for helping API users search through APIs.
+  Mica is built on Google and analyzes the content of Google results to
+  give more relevant results for programmers.
```

Comment:
how was this measured?

Reply:
They didn't seem to have a user study comparing Mica vs. plain google. But, they did briefly mention a formative study looking at how developers use google for learning APIs, which Mica addresses the problems they found. I added these details in the new draft